charts/agent/templates/_config-processors.tpl

```diff
@@ -81,3 +81,24 @@ memory_limiter:
 # needs to come after transform/watch_objects in pipelines
 observek8sattributes:
 {{- end -}}
+
+{{- define "config.processors.transform.observe_common"}}
+transform/observe_common:
+    error_mode: ignore
+    log_statements:
+      - context: log
+        statements:
+        # This is tricky, but essentially context:resource ensures that the
```

Comment:
I wonder if this might be an overkill.
I just checked `k8sattributes` processor, seems it can report `container.id`: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/k8sattributesprocessor/documentation.md

It's just not enabled by default. Can we try enabling that?
(And, if it works, we might as well use that for both logs and metrics.

Reply:
I saw that, but it says that it requires `k8s.container.restart_count` to guarantee the ID is correct, which is not available and it's not easy to get. One way is to expose it directly from the configs, but we obviously cannot rely on that.
[Others have the same problem](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/16431), and eventually you need to do the same "filename hack" to get the restart count so that k8sattributes can get the container id 😂 .
So we can just do the hack to get the ID directly I think

Reply:
> which is not available

Are you sure `k8s.container.restart_count` is not available? :)
checkout this worksheet's `resource_attributes` column :D 

https://102.observe-o2.com/workspace/41000009/log-explorer?datasetId=41051312&requiredCorrelationTags=k8s.pod.name&requiredCorrelationTags=k8s.cluster.name&requiredCorrelationTags=k8s.namespace.name&requiredCorrelationTags=k8s.pod.uid&s=1083-n5n1i77u

I guess the information is provided by OTel collector's built in log file processor, try follow this: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/beb7e34fd7c7ac08f5a9f2dae1f8fd618b50c69a/pkg/stanza/operator/parser/container/parser.go#L31

I checked, I couldn't find a single log line where we have a `k8s.container.name` without `k8s.container.restart_count`: https://102.observe-o2.com/workspace/41000009/log-explorer?datasetId=41051312&requiredCorrelationTags=k8s.pod.name&requiredCorrelationTags=k8s.cluster.name&requiredCorrelationTags=k8s.namespace.name&requiredCorrelationTags=k8s.pod.uid&s=1083-c4o13kaa

So, I think we can rely on it being there :)

Reply:
I see. I tried looking and I have no idea where this is set or which receiver/processor populates it. But indeed it's there in O2 in all logs ahah.
I will add it then.

Reply:
This is the row above. No `container.id` extracted despite matching with `restart_count` is possible.
<img width="1868" alt="image" src="https://github.com/user-attachments/assets/20032ac0-8c8f-4edc-95ec-90c6f12f883a">